templates/smtp-credentials.yaml

```diff
@@ -0,0 +1,272 @@
+    Type: AWS::IAM::Group
+    Properties:
+      GroupName: SMTPUserGroup
+
+  SmtpUser:
+    Type: AWS::IAM::User
+    Properties:
+      UserName: SMTPUser
+      Groups:
+        - !Ref SmtpUserGroup
+
+  SmtpUserPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: SMTPUserPolicy
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action: ses:SendRawEmail
+            Resource: '*'
+      Groups:
+        - !Ref SmtpUserGroup
+
+  SmtpUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      Serial: !Ref IamAccessKeyVersion
+      Status: Active
+      UserName: !Ref SmtpUser
+
+  SmtpPassword:
+    Type: Custom::SmtpPassword
+    Properties:
+      ServiceToken: !GetAtt SmtpLambdaFunction.Arn
+      Key: !GetAtt SmtpUserAccessKey.SecretAccessKey
+      ParameterType: password
+
+  SmtpUsername:
+    Type: Custom::SmtpUsername
+    Properties:
+      ServiceToken: !GetAtt SmtpLambdaFunction.Arn
+      Key: !Ref SmtpUserAccessKey
+      ParameterType: username
+
+  SmtpLambdaExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: SMTPLambdaExecutionRole
+      Description: Role assumed by Lambda to generate SMTP credentials
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: InlineSMTPLambdaExecutionRolePolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+                - Effect: Allow
+                  Action:
+                    - logs:CreateLogStream
+                    - logs:PutLogEvents
+                  Resource:
+                    - !Sub "arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/SMTPCredentialsLambdaFunction"
+                    - !Sub "arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/SMTPCredentialsLambdaFunction:log-stream:*"
+                - Effect: Allow
+                  Action:
+                    - ssm:PutParameter
+                    - ssm:DeleteParameter
+                  Resource:
+                    - !Sub "arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/smtp-username"
+                    - !Sub "arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/smtp-password"
+
+  SmtpKeyStack:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: !Sub ${TemplateRootUrl}/aws-infra/v0.2.19/KMS/kms-key.yaml
+      TimeoutInMinutes: 5
+      Parameters:
+        AliasName: 'alias/smtp-credentials'
+        AdminPrincipalArns: !Join
+          - ','
+          - - !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            - !ImportValue
+              'Fn::Sub': '${AWS::Region}-workflows-nextflow-ci-service-account-ServiceUserArn'
```

Comment:
you may want to give some humans access to this key?  otherwise you may need need to use the CI service account to debug problems with this.

Reply:
yes, was just dragging my feet on that hoping to be able to `!ImportValue`